web/packages/teleterm/src/ui/ConnectMyComputer/connectMyComputerContext.test.tsx

```diff
@@ -81,3 +121,110 @@ test('runAgentAndWaitForNodeToJoin re-throws errors that are thrown while spawni
+  const { appContext, rootCluster } = getMocksWithConnectMyComputerEnabled();
+  const { result } = renderHook(() => useConnectMyComputerContext(), {
+    wrapper: ({ children }) => (
+      <MockAppContextProvider appContext={appContext}>
+        <WorkspaceContextProvider value={null}>
+          <ConnectMyComputerContextProvider rootClusterUri={rootCluster.uri}>
+            {children}
+          </ConnectMyComputerContextProvider>
+        </WorkspaceContextProvider>
+      </MockAppContextProvider>
+    ),
+  });
+
+  await act(() => result.current.startAgent());
+
+  expect(
+    appContext.workspacesService.getConnectMyComputerAutoStartFlag(
+      rootCluster.uri
+    )
+  ).toBeTruthy();
+});
+
+test('killing the agent flips the workspace autoStart flag to false', async () => {
+  const { appContext, rootCluster } = getMocksWithConnectMyComputerEnabled();
+
+  const { result } = renderHook(() => useConnectMyComputerContext(), {
+    wrapper: ({ children }) => (
+      <MockAppContextProvider appContext={appContext}>
+        <WorkspaceContextProvider value={null}>
+          <ConnectMyComputerContextProvider rootClusterUri={rootCluster.uri}>
+            {children}
+          </ConnectMyComputerContextProvider>
+        </WorkspaceContextProvider>
+      </MockAppContextProvider>
+    ),
+  });
+
+  await act(() => result.current.killAgent());
+
+  expect(
+    appContext.workspacesService.getConnectMyComputerAutoStartFlag(
+      rootCluster.uri
+    )
+  ).toBeFalsy();
+});
+
+test('starts the agent automatically if the workspace autoStart flag is true', async () => {
+  const { appContext, rootCluster } = getMocksWithConnectMyComputerEnabled();
+  appContext.configService = createMockConfigService({
+    'feature.connectMyComputer': true,
+  });
+
+  const eventEmitter = new EventEmitter();
+  let currentAgentProcessState: AgentProcessState = {
+    status: 'not-started',
+  };
+  jest
+    .spyOn(appContext.mainProcessClient, 'getAgentState')
+    .mockImplementation(() => currentAgentProcessState);
+  jest
+    .spyOn(appContext.connectMyComputerService, 'isAgentConfigFileCreated')
+    .mockResolvedValue(true);
+  jest
+    .spyOn(appContext.connectMyComputerService, 'runAgent')
+    .mockImplementation(async () => {
+      currentAgentProcessState = { status: 'running' };
+      eventEmitter.emit('', currentAgentProcessState);
+    });
+  jest
+    .spyOn(appContext.connectMyComputerService, 'waitForNodeToJoin')
+    .mockResolvedValue(makeServer());
+  jest.spyOn(appContext.connectMyComputerService, 'downloadAgent');
+  jest
+    .spyOn(appContext.mainProcessClient, 'subscribeToAgentUpdate')
+    .mockImplementation((rootClusterUri, listener) => {
+      eventEmitter.on('', listener);
+      return { cleanup: () => eventEmitter.off('', listener) };
+    });
+  jest
+    .spyOn(appContext.workspacesService, 'getConnectMyComputerAutoStartFlag')
+    .mockReturnValue(true);
```

Comment:
Nit: I like to add whitespace before such setup. All of the spies before this one are indirect spies to make the test work, but this one is fundamental to the test.